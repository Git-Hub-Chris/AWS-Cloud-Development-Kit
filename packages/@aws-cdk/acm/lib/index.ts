export * from './certificate';
export * from './certificate-ref';

// The L1 library for AWS::CertificateManager:
export * from '../cfn/certificatemanager';
