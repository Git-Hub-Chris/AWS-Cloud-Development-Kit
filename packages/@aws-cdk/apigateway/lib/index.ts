// The L1 Library for AWS::ApiGateway:
export * from '../cfn/apigateway';
