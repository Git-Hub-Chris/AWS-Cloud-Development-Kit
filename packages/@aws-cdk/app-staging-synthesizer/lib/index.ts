export * from './default-staging-stack';
export * from './synthesizer';
