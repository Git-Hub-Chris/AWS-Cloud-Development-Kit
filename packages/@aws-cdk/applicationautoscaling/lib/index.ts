// The L1 Library for AWS::ApplicationAutoScaling:
export * from '../cfn/applicationautoscaling';
