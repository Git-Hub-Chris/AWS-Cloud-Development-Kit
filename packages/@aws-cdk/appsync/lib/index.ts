// The L1 Library for AWS::AppSync:
export * from '../cfn/appsync';
