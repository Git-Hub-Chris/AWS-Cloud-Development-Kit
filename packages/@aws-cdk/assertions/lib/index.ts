export * from './assertions';
export * from './matchers';