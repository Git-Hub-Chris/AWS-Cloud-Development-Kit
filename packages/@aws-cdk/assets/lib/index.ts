export * from './asset';
export { CopyOptions } from './fs/copy';
export * from './staging';
