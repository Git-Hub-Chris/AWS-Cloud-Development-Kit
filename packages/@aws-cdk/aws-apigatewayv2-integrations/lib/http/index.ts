export * from './base-types';
export * from './alb';
export * from './aws-proxy';
export * from './nlb';
export * from './service-discovery';
export * from './http-proxy';
export * from './lambda';
export * from './mapping-expression';
