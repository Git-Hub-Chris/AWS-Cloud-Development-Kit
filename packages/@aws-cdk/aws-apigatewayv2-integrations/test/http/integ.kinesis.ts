import { HttpApi } from '@aws-cdk/aws-apigatewayv2';
import { PolicyStatement, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
import { Stream } from '@aws-cdk/aws-kinesis';
import { App, CfnOutput, Stack } from '@aws-cdk/core';
import { KinesisPutRecordIntegration, Mapping, StreamMappingExpression, StringMappingExpression } from '../../lib';

// Stack verification steps
// <step-1> deploy the stack and find the api URL and the kinesis stream name
// <step-2> make a POST request to the URL, with a message body
// verify that the message has been written to the kinesis stream:
// <step-3> aws kinesis get-shard-iterator --stream-name <stream name> --shard-id 0 --shard-iterator-type TRIM_HORIZON
// <step-4> aws kinesis get-records --shard-iterator <shard-iterator>
// The message will be base64 encoded in the Data attribute of the record
const app = new App();
const stack = new Stack(app, 'integ-kinesis');
const httpApi = new HttpApi(stack, 'IntegKinesisApi');
const stream = new Stream(stack, 'Stream');
const role = new Role(stack, 'Role', {
  assumedBy: new ServicePrincipal('apigateway.amazonaws.com'),
});
role.addToPrincipalPolicy(new PolicyStatement({
  actions: ['kinesis:PutRecord'],
  resources: [stream.streamArn],
}));
httpApi.addRoutes({
  path: '/',
  integration: new KinesisPutRecordIntegration({
    data: StringMappingExpression.fromMapping(Mapping.fromRequestBody()),
    stream: StreamMappingExpression.fromStream(stream),
    partitionKey: StringMappingExpression.fromValue('1'),
    role,
  }),
});

new CfnOutput(stack, 'ApiUrl', {
  value: httpApi.apiEndpoint,
});
new CfnOutput(stack, 'StreamName', {
  value: stream.streamName,
});
