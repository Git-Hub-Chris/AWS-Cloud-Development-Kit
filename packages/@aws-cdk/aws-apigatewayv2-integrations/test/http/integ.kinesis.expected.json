{
  "Resources": {
    "IntegKinesisApi890BF27F": {
      "Type": "AWS::ApiGatewayV2::Api",
      "Properties": {
        "Name": "IntegKinesisApi",
        "ProtocolType": "HTTP"
      }
    },
    "IntegKinesisApiDefaultStage245E3852": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "IntegKinesisApi890BF27F"
        },
        "StageName": "$default",
        "AutoDeploy": true
      }
    },
    "IntegKinesisApiANYHttpIntegrationc304c3d15ebb729a043fc95e653032645012054C": {
      "Type": "AWS::ApiGatewayV2::Integration",
      "Properties": {
        "ApiId": {
          "Ref": "IntegKinesisApi890BF27F"
        },
        "IntegrationType": "AWS_PROXY",
        "CredentialsArn": {
          "Fn::GetAtt": [
            "Role1ABCC5F0",
            "Arn"
          ]
        },
        "IntegrationSubtype": "Kinesis-PutRecord",
        "PayloadFormatVersion": "1.0",
        "RequestParameters": {
          "StreamName": {
            "Ref": "Stream790BDEE4"
          },
          "Data": "$request.body",
          "PartitionKey": "1"
        }
      }
    },
    "IntegKinesisApiANY248F7340": {
      "Type": "AWS::ApiGatewayV2::Route",
      "Properties": {
        "ApiId": {
          "Ref": "IntegKinesisApi890BF27F"
        },
        "RouteKey": "ANY /",
        "AuthorizationType": "NONE",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "IntegKinesisApiANYHttpIntegrationc304c3d15ebb729a043fc95e653032645012054C"
              }
            ]
          ]
        }
      }
    },
    "Stream790BDEE4": {
      "Type": "AWS::Kinesis::Stream",
      "Properties": {
        "ShardCount": 1,
        "RetentionPeriodHours": 24,
        "StreamEncryption": {
          "Fn::If": [
            "AwsCdkKinesisEncryptedStreamsUnsupportedRegions",
            {
              "Ref": "AWS::NoValue"
            },
            {
              "EncryptionType": "KMS",
              "KeyId": "alias/aws/kinesis"
            }
          ]
        }
      }
    },
    "Role1ABCC5F0": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        }
      }
    },
    "RoleDefaultPolicy5FFB7DAB": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "kinesis:PutRecord",
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "Stream790BDEE4",
                  "Arn"
                ]
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "PolicyName": "RoleDefaultPolicy5FFB7DAB",
        "Roles": [
          {
            "Ref": "Role1ABCC5F0"
          }
        ]
      }
    }
  },
  "Conditions": {
    "AwsCdkKinesisEncryptedStreamsUnsupportedRegions": {
      "Fn::Or": [
        {
          "Fn::Equals": [
            {
              "Ref": "AWS::Region"
            },
            "cn-north-1"
          ]
        },
        {
          "Fn::Equals": [
            {
              "Ref": "AWS::Region"
            },
            "cn-northwest-1"
          ]
        }
      ]
    }
  },
  "Outputs": {
    "ApiUrl": {
      "Value": {
        "Fn::GetAtt": [
          "IntegKinesisApi890BF27F",
          "ApiEndpoint"
        ]
      }
    },
    "StreamName": {
      "Value": {
        "Ref": "Stream790BDEE4"
      }
    }
  }
}